crates/viewer/re_viewer_context/src/image_info.rs

```diff
@@ -165,6 +165,179 @@ impl ImageInfo {
             Cow::Owned(dest)
         }
     }
+
+    /// Best-effort.
+    ///
+    /// `u8` and `u16` images are returned as is.
+    ///
+    /// Other data types are remapped from the given `data_range`
+    /// to the full `u16` range, then rounded.
+    ///
+    /// Returns `None` for invalid images (if the buffer is the wrong size).
+    pub fn to_dynamic_image(&self, data_range: RangeInclusive<f32>) -> Option<image::DynamicImage> {
+        re_tracing::profile_function!();
+
+        use image::{DynamicImage, GrayImage, RgbImage, RgbaImage};
+        type Gray16Image = image::ImageBuffer<image::Luma<u16>, Vec<u16>>;
+        type Rgb16Image = image::ImageBuffer<image::Rgb<u16>, Vec<u16>>;
+        type Rgba16Image = image::ImageBuffer<image::Rgba<u16>, Vec<u16>>;
+
+        let (w, h) = (self.width(), self.height());
+
+        if let Some(pixel_format) = self.format.pixel_format {
+            // Convert to RGB:
+            let buf: &[u8] = &self.buffer;
+            let mut rgb = Vec::with_capacity((w * h * 3) as usize);
+            for y in 0..h {
+                for x in 0..w {
+                    // NOTE: the name `y` is already taken for the coordinate, so we use `luma` here.
+                    let [luma, u, v] = match pixel_format {
+                        PixelFormat::NV12 => {
```

Comment:
Duplication of the decoding makes me very uneasy. I'd rather we divide above's `ImageInfo::from_xyz` to get colors as well and then use that again. Otherwise, we'll keep on adding duplicated decoding logic.

Or easier just split this out into a standalone `color_at_postion_for_pixel_format` function that takes a blob, format and coordinate which you then can call here and above

Reply:
I unified it now, but at the cost of making it harder to add a new `PixelFormat` that is not chroma-downsampled. I'll leave it for the next person to improve.